src/nvim/api/buffer.c

```diff
@@ -441,6 +442,8 @@ void nvim_buf_set_lines(uint64_t channel_id,
     goto end;
   }
 
+  bcount_t deleted_bytes = MAX(get_region_bytecount(start, end, 0, 0)-1, 0);
```

Comment:
this doesn't seem right. `get_region_bytecount(start, end, 0, 0)`  should return the actual size always (perhaps with the change I suggested below).